nucleo_h745zi_q add fdcan1

KozhinovAlexander opened the pull request:
add fdcan1

KozhinovAlexander committed:
boards: arm: nucleo_h745zi_q

add fdcan1, fdcan2 and usb_fs DT configurations

Signed-off-by: Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>

erwango commented on boards/st/nucleo_h745zi_q/nucleo_h745zi_q_stm32h745xx_m7.dts:
```diff
@@ -144,3 +144,27 @@ zephyr_udc0: &usbotg_fs {
+		     &usb_otg_fs_dm_pa11
+		     &usb_otg_fs_dp_pa12>;
+	pinctrl-names = "default";
+	status = "disabled";
```
Nodes added in boards should all be enabled.

KozhinovAlexander replied:
fdcan1 conflicts with i2c1 pinouts. Moving fdcan1 to another subset of pins will not work with waveshark rs485-can hat matching perfectly on the board. Thus I have to disable fdcan1 since i2c1 was as first one there. I've also added a comment, what needed to be done to enable fdcan1.

KozhinovAlexander replied:
fdcan1 will become pins mentioned by @FRASTM

erwango reviewed (changes requested):
Thanks for the contribution, please also fix CI.

KozhinovAlexander commented:
> Thanks for the contribution, please also fix CI.

I've updated my PR and fixed CI. There is only one open point regarding fdcan1 status. Please take a look on it in corresponding conversation above.

erwango commented on boards/st/nucleo_h745zi_q/nucleo_h745zi_q_stm32h745xx_m7.yaml:
```diff
@@ -16,6 +16,7 @@ supported:
   - counter
   - i2c
   - pwm
+  - can
```
By adding can in the list, it will be tested automatically, which will fail as node is disabled by default.

KozhinovAlexander replied:
I will change to pins set mentioned by @FRASTM

erwango commented on boards/st/nucleo_h745zi_q/nucleo_h745zi_q_stm32h745xx_m7.dts:
```diff
@@ -144,3 +144,13 @@ zephyr_udc0: &usbotg_fs {
+
+&fdcan1 {
+	pinctrl-0 = <&fdcan1_rx_pb8 &fdcan1_tx_pb9>;
+	// HSE will be used by default. Uncomment below to enable APB1.2 120MHz clock
+	// clocks = <&rcc STM32_CLOCK_BUS_APB1_2 0x00000100>,
+	//	    <&rcc STM32_SRC_PLL1_Q FDCAN_SEL(1)>;
+	pinctrl-names = "default";
+	// pins are confilcting with i2c1 - disable i2c1 first to enable
```
This is fine to me, but please use C style comments.

FRASTM replied:
can you please elaborate ; 
the board schematics and the userManual (Table 17. ) gives on connector CN9
CAN_RX PD0 CAN_1 
CAN_TX PD1 CAN_1 
So changing this configuration will not only conflict with I2C 1 pinning but it will also mismatch the nucleo board specification.

KozhinovAlexander replied:
Thank you both for your valuable comments. 
@erwango I will change to C style comments.

@FRASTM  You're right. I've choosen fdcan1 pins that way, cause I am using [waveshare's rs485-can shield](https://www.waveshare.com/rs485-can-shield.htm), since it it only shield (i know), which can be plugged to almost any nucle board. Thus very handy. 
But it is true - it will make more sense in the current context to move to pins set mentioned by you, s.t. can can be enabled by default and tests will run. Afterwards the specific projects may alter pins for own need e.g. waveshare shield in my case.

erwango commented on boards/st/nucleo_h745zi_q/doc/index.rst:
```diff
@@ -112,6 +112,8 @@ features:
 +-------------+------------+-------------------------------------+
 | SPI         | on-chip    | spi                                 |
 +-------------+------------+-------------------------------------+
+| FDCAN       | on-chip    | CAN-FD Control Area Network         |
```
A note in the doc explaining the conflict and disabled state by default would be nice (as some people don't check dts files)

KozhinovAlexander replied:
I will change to pins set mentioned by @FRASTM and enable it then.

KozhinovAlexander commented:
@erwango @FRASTM it shall be ready now. Please take a look on it.
